### PR TITLE
SIMSBIOHUB-148: Create New Agency Database Table to Support All Agencies

### PR DESCRIPTION
**Pull request by NickPhura**
## Links to Jira Tickets

https://apps.nrs.gov.bc.ca/int/jira/browse/SIMSBIOHUB-148

### Funding Source -> Agency
- New migration to update existing `funding_source` table to `agency`.
  - Update columns, indexes, constraints, sequences, etc that used `funding_source` in their name. Rename to use `agency`.
  - Inserts new agency values (from static coordinator agency list)
- Removed static coordinator agency list
- Update `tr_project_funding_source` trigger to reference renamed `agency` table and columns.
- Rename instances of functions/variables that use `funding_source` to use `agency`
  - Renamed/expanded some shortened variable names `fs` (aka: `funding_source`) to `agency`
  - Related: expanded some other related shortened variable names `pfs` to `project_funding_source` for clarity
- Small fix for issue in region-service in Dev (including in this PR for simplicity)

### Other
- Add migration to update miss-spelled triggers for region tables (added in from a previous PR: https://github.com/bcgov/biohubbc/pull/1051).

### Documentation
- Updated database README to add documentation on database naming conventions, and sample migrations, and things to consider when writing a migration, etc.

## Testing Notes

- Focus testing effort on anything related to Funding (project and survey level).
  - The app should function the exact same as before.

### CODE REVIEW

**Review by curtisupshall — DISMISSED** [review]
Code looks good!

File: database/src/migrations/20230719000000_update_funding_sources.ts [hunk 1]
```diff
@@ -0,0 +1,625 @@
...
+  INSERT INTO INVESTMENT_ACTION_CATEGORY (agency_id, record_effective_date, name) VALUES ((SELECT agency_id FROM agency WHERE name = 'Michigan State University'), NOW(), 'Not Applicable');
+  INSERT INTO INVESTMENT_ACTION_CATEGORY (agency_id, record_effective_date, name) VALUES ((SELECT agency_id FROM agency WHERE name = 'Mid Vancouver Island Habitat Enhancement Society'), NOW(), 'Not Applicable');
+  INSERT INTO INVESTMENT_ACTION_CATEGORY (agency_id, record_effective_date, name) VALUES ((SELECT agency_id FROM agency WHERE name = 'Ministry of Environment & Climate Change Strategy'), NOW(), 'Not Applicable');
+  INSERT INTO INVESTMENT_ACTION_CATEGORY (agency_id, record_effective_date, name) VALUES ((SELECT agency_id FROM agency WHERE name = 'Ministry of Forests'), NOW(), 'Not Applicable');
+  INSERT INTO INVESTMENT_ACTION_CATEGORY (agency_id, record_effective_date, name) VALUES ((SELECT agency_id FROM agency WHERE name = 'Ministry of Forests, Lands, Natural Resource Operations & Rural Development'), NOW(), 'Not   Applicable');
```
Comment (al-rosenthal) [inline database/src/migrations/20230719000000_update_funding_sources.ts]: nit: there's an extra space in 'Not  Applicable'

**Review by al-rosenthal — DISMISSED** [review]
ran locally and things are working as expected. Great examples in the read me

File: database/src/migrations/20230719000000_update_funding_sources.ts (hunk 1 above)
Comment (NickPhura) [reply database/src/migrations/20230719000000_update_funding_sources.ts]: fixed!

**Review by al-rosenthal — APPROVED** [review]
🦖

**Review by KjartanE — APPROVED** [review]
Ran locally, works as intended! 😹

**Pull request merged**